### DISCUSSION TITLE
task instance is null

### DISCUSSION THREAD

**Discussion by penguy-up** [apache/dolphinscheduler#12508] [Q&A] [answered]
版本为3.1，调度失败，ui提示task instance is null，该如何调整，保证热任务正常调度
![image](https://user-images.githubusercontent.com/33483797/197441957-673ccaca-7805-4dd2-9058-5996f91fc235.png)

**Comment by DarkAssassinator** [comment, marked as answer]
hi @Kylin-9, this error caused by cannot find the host column in `t_ds_task_instance` for this record.

**Reply by penguy-up** [reply]
Why is this happening and how should I fix it? thanks

**Reply by penguy-up** [reply]
Manually triggering execution can assign hosts, but automatic scheduling cannot

**Reply by penguy-up** [reply]
The test found that when the task parameters are not configured, the scheduling is executed normally, but the parameters are configured, and the scheduling fails : task instance is null
![image](https://user-images.githubusercontent.com/33483797/197995217-24e9b6b5-82e9-4b3e-9d0b-e7c3711bc9ba.png)

**Comment by penguy-up** [comment]
@DarkAssassinator

**Reply by DarkAssassinator** [reply]
> @DarkAssassinator

hi @Kylin-9, so sorry for the late reply. So u mean that auto scheduling will not set the host column, right?

**Comment by ruanwenjun** [comment]
@Kylin-9 Could you please provide the master log related to the task instance.

**Comment by DarkAssassinator** [comment]
hi @Kylin-9  After i check the source code, this issue caused by null process commandparam, and it has fixed, please check https://github.com/apache/dolphinscheduler/pull/12419
